### ISSUE TITLE
Reimplement WAF map

### ISSUE THREAD

**Issue by jptosso** [corazawaf/libcoraza#29]
Although the general suggestion is a sync.Map, we must make sure it is scalable in terms of resources and it properly cleans memory for deleted instances. 

Also we should stop storing the WAF itself and we should store both, WAFConfig and WAF in a single struct, that way we can extend configuration for configuration merging, which would be useful for nginx and apache.

```go
type waf struct {
    WAF coraza.WAF
    Config coraza.WAFConfig
}
```

**Comment by potats0** [comment]
you could expose seclang.FromString api for updated

**Comment by jptosso** [comment]
No, this is now first class within the coraza merge experimental feature
